dyninstAPI_RT/src/RTsignal.c

```diff
@@ -65,7 +64,11 @@ static signal_type real_signal = NULL;
 dynsighandler_t user_trap_handler = NULL;
 DLLEXPORT dynsighandler_t signal(int signum, dynsighandler_t handler) {
     if (!real_signal) {
-        real_signal = (signal_type)dlsym(RTLD_NEXT, "signal");
+        void* libc_handle = dlopen("libc.so.6", RTLD_LAZY);
```

Comment:
@ManojKumarChauhan Can you explain why you changed dlsym from using RTLD_NEXT to directly specifying the library name? Directly dlsym a specific library is likely to break other tools if they run at the same time with a dyninst instrumented binary. 

RTLD_NEXT will ensure that dyninst's wrapper will call other tools' wrappers.

Reply:
I found issue in dynamic instrumentation while using this code. I just reverted it back which was working fine.